src/EnergyPlus/AirTerminalUnit.hh

```diff
@@ -0,0 +1,118 @@
+#ifndef AirTerminalUnit_hh_INCLUDED
+#define AirTerminalUnit_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+
+namespace EnergyPlus {
+
+// types of air terminal units, refactored from old DataDefineEquip
+enum AirTerminalUnitType {
+	notYetDetermined,
+	dualDuctConstVolume,
+	dualDuctVAV,
+	singleDuctVAVReheat,
+	singleDuctConstVolReheat,
+	singleDuctVAVNoReheat,
+	singleDuct_SeriesPIU_Reheat,
+	singleDuct_ParallelPIU_Reheat,
+	singleDuct_ConstVol_4PipeInduc,
+	singleDuctVAVReheatVSFan,
+	singleDuctCBVAVReheat,
+	singleDuctCBVAVNoReheat,
+	singleDuctConstVolCooledBeam,
+	dualDuctVAVOutdoorAir,
+	singleDuctUserDefined,
+	singleDuctInletATMixer,
+	singleDuctSupplyATMixer,
+	singleDuctConstVolFourPipeBeam
+};
+
+// base class for all air distribution units.  zone air terminals for connecting to central air handlers
+class AirTerminalUnit
+{
+
+protected: // Creation
+
+	// Default Constructor
+	AirTerminalUnit() :
+		terminalType( notYetDetermined ),
+		aDUNum( 0 ),
+		airAvailSchedNum( 0 ),
+		airAvailable( false ),
+		vDotDesignPrimAir( 0.0 ),
+		vDotDesignPrimAirWasAutosized( false ),
+		mDotDesignPrimAir( 0.0 ),
+		airInNodeNum( 0 ),
+		airOutNodeNum( 0 ),
+		zoneIndex( 0 ),
+		zoneNodeIndex( 0 )
+	{}
+
+	// Copy Constructor
+	AirTerminalUnit( AirTerminalUnit const & ) = default;
+
+	// Move Constructor
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit( AirTerminalUnit && ) = default;
+#endif
+
+public: // Creation
+
+	// Destructor
+	virtual
+	~AirTerminalUnit()
+	{}
+
+protected: // Assignment
+
+	// Copy Assignment
+
+	AirTerminalUnit &
+	operator =( AirTerminalUnit const & ) = default;
+
+	// Move Assignment
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit &
+	operator =( AirTerminalUnit && ) = default;
+#endif
+
+public: // Methods		REMOVE ANY OF THESE THAT AREN'T COMMON (WITH SAME ARGS) TO ALL SUB-TYPES
+
+	// for unit tests
+	virtual
+	void
+	clear_state() = 0;
+
+	virtual
+	void
+	simulate(
+		bool const FirstHVACIteration, // TRUE if first HVAC iteration in time step
+		Real64 & NonAirSysOutput // convective cooling by the beam system [W]
+	) = 0;
+
+
+
+public: // Data
```

Comment:
@EnergyArchmage I guess I should be more specific. Does anything outside of the derived classes need to access that data within EnergyPlus simulation, not unit tests?

I personally think making things public to make unit testing easier isn't the best design. The member functions and variables that are public should be thought of as a contract with users of your class. These are the public API of your class and should only be changed when the cost of not changing them is greater than the cost of changing them. Anything else is a private implementation detail and should not be exposed. This allows you to change implementations without affecting the public API. 

This starts to get into unit testing paradigms but unit tests should usually only test public APIs. If something is private/protected and needs to be unit tested for valid reasons (very complex, critical but not user facing function, etc) then you can `friend` a Fixture. I've had to do this for SQLiteProcedures and my process_idd() function. But other people might have different opinions.

All that said, if these member variables truly need to be public then all I said does not apply in this case.